front/lib/actions/mcp_internal_actions/servers/hubspot.ts

```diff
@@ -35,362 +37,136 @@ const createServer = (auth: Authenticator, mcpServerId: string): McpServer => {
     "get_object_properties",
     "Lists all available properties for a Hubspot object. When creatableOnly is true, returns only properties that can be modified through forms (excludes hidden, calculated, read-only and file upload fields).",
     {
-      objectType: z.enum(SUPPORTED_OBJECT_TYPES_READ),
+      objectType: z.enum(ALL_OBJECTS),
       creatableOnly: z.boolean().optional(),
     },
     async ({ objectType, creatableOnly = true }) => {
-      const accessToken = await getAccessTokenForInternalMCPServer(auth, {
-        mcpServerId,
-      });
-      if (!accessToken) {
-        return {
-          isError: true,
-          content: [{ type: "text", text: "No access token found" }],
-        };
-      }
-
-      try {
-        const properties = await getObjectProperties({
-          accessToken,
-          objectType,
-          creatableOnly,
-        });
-        const content = JSON.stringify(properties, null, 2);
-
-        return {
-          isError: false,
-          content: [
-            { type: "text", text: "Object properties fetched successfully" },
-            { type: "text", text: content },
-          ],
-        };
-      } catch (error) {
-        logger.error(
-          {
-            error,
-            objectType,
-            mcpServerId,
-            server: "hubspot",
-          },
-          "[Hubspot MCP Server] Error fetching object properties."
-        );
-        return {
-          isError: true,
-          content: [{ type: "text", text: "Error fetching object properties" }],
-        };
-      }
+      return withAuth(auth, mcpServerId, (accessToken) =>
+        getObjectProperties({ accessToken, objectType, creatableOnly })
+      );
     }
   );
 
   server.tool(
     "create_object",
-    `Creates a new object in Hubspot. Supports ${SUPPORTED_OBJECT_TYPES_WRITE.join(", ")}.`,
+    `Creates a new object in Hubspot. Supports ${SIMPLE_OBJECTS.join(", ")}.`,
```

Comment:
Do we need to have the enum spelled out in the description if already in the type ? Is the enum type being rendered to the model ?

Reply:
It's actually for the users! 😄 
 
<img width="586" alt="Screenshot 2025-04-25 at 15 41 38" src="https://github.com/user-attachments/assets/91086756-58df-46aa-947a-609a915826a0" />